trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/signing/SigningController.java

```diff
@@ -94,14 +95,25 @@ public SigningMetadata validateAndParseAuthorization(Request request, SigningSer
 
         return credentialsController.withCredentials(parsedAuthorization.accessKey(), session, credentials -> {
             SigningMetadata metadata = new SigningMetadata(signingServiceType, credentials, session, parsedAuthorization.region(), request.requestContent());
-            if (isValidAuthorization(metadata, Credentials::emulated, parsedAuthorization, request.requestUri(), request.requestHeaders(), request.requestQueryParameters(), request.httpVerb())) {
-                return Optional.of(metadata);
-            }
-            return Optional.empty();
+            return isValidAuthorization(metadata, Credentials::emulated, parsedAuthorization, request.requestUri(), request.requestHeaders(), request.requestQueryParameters(), request.httpVerb());
         }).orElseThrow(() -> new WebApplicationException(Response.Status.UNAUTHORIZED));
     }
 
-    private String internalSignRequest(
+    public ChunkSigningSession startChunkSigningSession(
```

Comment:
This function seems unused? Will this PR deal with enabling chunk signature verification or just the skeleton for it?

Reply:
Oof - bad rebase. I'll fix it.